pkg/chaosdaemon/tc_server.go

```diff
@@ -148,3 +153,268 @@ func generateQdiscArgs(action string, qdisc *pb.Qdisc) ([]string, error) {
 
 	return args, nil
 }
+
+func (s *daemonServer) SetTcs(ctx context.Context, in *pb.TcsRequest) (*empty.Empty, error) {
+	log.Info("handling tc request", "tcs", in)
+
+	pid, err := s.crClient.GetPidFromContainerID(ctx, in.ContainerId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "get pid from containerID error: %v", err)
+	}
+	nsPath := GetNsPath(pid, netNS)
+	client := buildTcClient(ctx, nsPath)
+	err = client.flush()
+	if err != nil {
+		return &empty.Empty{}, err
+	}
+
+	globalTc := []*pb.Tc{}
+	filterTc := map[string][]*pb.Tc{}
+
+	for _, tc := range in.Tcs {
+		if tc.Ipset == "" {
+			globalTc = append(globalTc, tc)
+		} else {
+			// TODO: support multiple tc with one ipset
+			filterTc[tc.Ipset] = append(filterTc[tc.Ipset], tc)
+		}
+	}
+
+	for index, tc := range globalTc {
+		parentArg := "root"
+		if index > 0 {
+			parentArg = fmt.Sprintf("parent %d:", index)
+		}
+
+		handleArg := fmt.Sprintf("handle %d:", index+1)
+
+		err := client.addTc(parentArg, handleArg, tc)
+		if err != nil {
+			log.Error(err, "error while adding tc")
+			return &empty.Empty{}, err
+		}
+	}
+
+	parent := len(globalTc)
+	band := 3 + len(filterTc)
```

Comment:
why need to add `3` at here?

Reply:
Because 3 sfq qdisc is needed for packets that cannot fit in any filter. Comment has been added here.